students/briggsm/session06/mailroom4/mailroom4_test.py

```diff
@@ -0,0 +1,74 @@
+'''Unit Test for mailroom4'''
+
+import json
+import mock
+import mailroom4 as m4
+
+persistantdata = "mailroomdata_try.json"
+maildata = m4.open_json(persistantdata)
+
+def test_open_json():
+    '''Opens a JSON file and returns the JSON data as a dictionary.'''
+    file = open("mailroomdata.json")
+    expected = '{"John Dill": [10.0, 20.0, 30.0], "Sarah Pickle": [10.0, 20.0, 30.0], "Ornett Salt": [10.0, 20.0, 30.0], "Fanny Pepper": [10.0, 20.0, 30.0], "Jo Fang": [50.0]}'
+    assert expected==file.read()
+
+
+def test_write_json():
+    '''Write dictionary as JSON to the target filename.'''
+    jsondata_dict = {"test":"value"}
+    jsondata_test = str(jsondata_dict)
+    filename_test = "test.json"
+    testout = open(filename_test, "w")
+    testout.write(jsondata_test)
+    testout.close()
+    file = open(filename_test)
+    assert jsondata_test == file.read()
```

Comment:
careful testing for an exact match -- maybe test if, when read, the json has the right data it it. 

why?

JSON can have arbitrary whitespace

the order of keys is not guaranteed.

So it would be correct, but not be exactly the same.